RequirementMachine: Leave behind conflicting requirements in the minimized signature

slavapestov opened the pull request:
Requirement lowering only expects that it won't see two requirements
of the same kind (except for conformance requirements). So only mark
those as conflicting.

This addresses a crash-on-invalid and improves diagnostics for
move-only generics, because a conflict won't drop the copyability
of a generic parameter and expose a move-only-naive user to
confusing error messages.

Fixes https://github.com/apple/swift/issues/61031.
Fixes https://github.com/apple/swift/issues/63997.
Fixes rdar://problem/111991454.

AnthonyLatsis commented on validation-test/compiler_crashers_2_fixed/issue61031.swift:
```diff
@@ -0,0 +1,16 @@
+// RUN: %target-typecheck-verify-swift
```
Do you mind sticking to the `issue-*` format for file names? This is the one I'm planning to use once I start renaming all the `sr*` files.

slavapestov committed:
RequirementMachine: Leave behind conflicting requirements in the minimized signature

Requirement lowering only expects that it won't see two requirements
of the same kind (except for conformance requirements). So only mark
those as conflicting.

This addresses a crash-on-invalid and improves diagnostics for
move-only generics, because a conflict won't drop the copyability
of a generic parameter and expose a move-only-naive user to
confusing error messages.

Fixes #61031.
Fixes #63997.
Fixes rdar://problem/111991454.

slavapestov commented:
@swift-ci Please smoke test

ahoppen commented on test/IDE/complete_where_clause.swift:
```diff
@@ -234,8 +234,11 @@ extension WithAssoc where T.#^EXT_ASSOC_MEMBER_1^#
+// shows members of 'Int'.
 extension WithAssoc where Int == T.#^EXT_ASSOC_MEMBER_2^# 
-// Same as EXT_ASSOC_MEMBER
+// EXT_ASSOC_MEMBER_2: Begin completions, {{.*}} items
+// EXT_ASSOC_MEMBER_2: Keyword/None:                           Type[#Int.Type#];
```
This is a regression. Do you know how it used to get parsed or why the behavior might have changed?
